src/videoConfProvider.ts

```diff
@@ -133,15 +143,42 @@ export class JitsiProvider implements IVideoConfProvider {
 							name: user.name,
 							avatar: await this.getAbsoluteUrl(`avatar/${user.username}`),
 							email: `user_${user._id}@rocket.chat`,
+							...(this.useJaaS && { moderator: user?._id === call.createdBy._id }),
 						},
 				  }
 				: '',
 		};
 
-		if (user && user._id === call.createdBy._id) {
+		if (user && user._id === call.createdBy._id && !this.useJaaS) {
 			payload.moderator = true;
 		}
 
+		if (this.useJaaS) {
+			const jaasHeader = {
+				...header,
+				kid: this.jaasApiKey,
+				alg: 'RS256',
+			};
+
+			const jaasPayload = {
+				...payload,
+				iss: 'chat',
+				sub: this.jitsiAppId,
+				aud: 'jitsi',
+				context: {
+					...(payload.context || {}),
+					features: {},
+				},
+			};
+
+			return jws.JWS.sign(
+				'RS256',
+				JSON.stringify(jaasHeader),
+				JSON.stringify(jaasPayload),
+				new Buffer(this.jaasPrivateKey, 'base64').toString('utf8'),
```

Comment:
Maybe good to fix:

The `Buffer` constructor is deprecated due to security concerns. Specifically, calling `new Buffer(arg)` could expose uninitialized memory if `arg` is a number. This is a potential security vulnerability in the application if an attacker can control the value of `arg`. 

In our case, we are using `new Buffer(this.jaasPrivateKey, 'base64')`, which is not directly a security concern because we're not passing a number. However, the Buffer constructor is still deprecated.

Can be replaced with something like:
```
Buffer.from(this.jaasPrivateKey, 'base64').toString('utf8'),
```

Thank you!